src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstance.cs

```diff
@@ -70,12 +72,20 @@ public class VisualStudioInstance
         /// </summary>
         public string InstallationPath { get; }
 
-        public VisualStudioInstance(Process hostProcess, DTE dte, ImmutableHashSet<string> supportedPackageIds, string installationPath)
+        public string DumpDirectoryPath { get; }
+
+        private bool _exceptionHappened;
+
+        public VisualStudioInstance(Process hostProcess, DTE dte, ImmutableHashSet<string> supportedPackageIds, string installationPath, string dumpDirectoryPath)
         {
             HostProcess = hostProcess;
             Dte = dte;
             SupportedPackageIds = supportedPackageIds;
             InstallationPath = installationPath;
+            DumpDirectoryPath = dumpDirectoryPath;
+
+            _exceptionHappened = false;
+            AppDomain.CurrentDomain.FirstChanceException += FirstChanceExceptionHandler;
```

Comment:
This is assigning a new handler every time we create an instance of this type.  Seems wrong.  Feel like this needs to be done more centrally so that it's done only once per AppDomain instance.